src/urllib3/connection.py

```diff
@@ -58,6 +59,8 @@ class ConnectionError(Exception):
 # (ie test_recent_date is failing) update it to ~6 months before the current date.
 RECENT_DATE = datetime.date(2019, 1, 1)
 
+_CONTAINS_CONTROL_CHAR_RE = re.compile("[\x00-\x20\x7f]")
```

Comment:
RFC 2730 mentions that the method is a token, and that a token is a number of VCHAR (US-ASCII visible characters):

```
     method         = token
     token          = 1*tchar

     tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                    / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
                    / DIGIT / ALPHA
                    ; any VCHAR, except delimiters
```

[h11 defines token](https://github.com/python-hyper/h11/blob/68e32db/h11/_abnf.py) as `token = r"[-!#$%&'*+.^_`|~0-9a-zA-Z]+"`, so maybe we can steal that regex? It would have prevent the attack mentioned in the bug.

Reply:
Good idea, I'll copy h11.